hack/addData/addData.go

```diff
@@ -0,0 +1,751 @@
+		"c7i":        "48xlarge",
+		"hpc6a":      "48xlarge",
+		"hpc7g":      "16xlarge",
+		"hpc7a":      "96xlarge",
+		"r5":         "metal",
+		"r5a":        "24xlarge",
+		"r5ad":       "24xlarge",
+		"r5b":        "metal",
+		"r5d":        "metal",
+		"r5dn":       "metal",
+		"r5n":        "metal",
+		"r6a":        "metal",
+		"r6g":        "metal",
+		"r6gd":       "metal",
+		"r6i":        "metal",
+		"r6idn":      "metal",
+		"r6in":       "metal",
+		"r6id":       "metal",
+		"r7a":        "metal-48xl",
+		"r7g":        "metal",
+		"r7gd":       "16xlarge",
+		"r7iz":       "32xlarge",
+		"u-3tb1":     "56xlarge",
+		"u-6tb1":     "metal",
+		"u-9tb1":     "metal",
+		"u-12tb1":    "metal",
+		"u-18tb1":    "metal",
+		"u-24tb1":    "metal",
+		"x1":         "32xlarge",
+		"x2gd":       "metal",
+		"x2idn":      "metal",
+		"x2iedn":     "metal",
+		"x2iezn":     "metal",
+		"x1e":        "32xlarge",
+		"z1d":        "metal",
+		"d3":         "8xlarge",
+		"d3en":       "12xlarge",
+		"h1":         "16xlarge",
+		"i3":         "metal",
+		"i3en":       "metal",
+		"i4g":        "16xlarge",
+		"i4i":        "metal",
+		"im4gn":      "16xlarge",
+		"is4gen":     "8xlarge",
+		"f1":         "16xlarge",
+		"g3":         "16xlarge",
+		"g4ad":       "16xlarge",
+		"g4dn":       "metal",
+		"g5":         "48xlarge",
+		"g5g":        "metal",
+		"inf1":       "24xlarge",
+		"inf2":       "48xlarge",
+		"p2":         "16xlarge",
+		"p3":         "16xlarge",
+		"p3dn":       "24xlarge",
+		"p4d":        "24xlarge",
+		"p4de":       "24xlarge",
+		"p5":         "48xlarge",
+		"trn1":       "32xlarge",
+		"trn1n":      "32xlarge",
+		"vt1":        "24xlarge",
+		"a1":         "metal",
+		"c1":         "xlarge",
+		"c3":         "8xlarge",
+		"c4":         "8xlarge",
+		"g2":         "8xlarge",
+		"i2":         "8xlarge",
+		"m1":         "xlarge",
+		"m2":         "4xlarge",
+		"m3":         "2xlarge",
+		"m4":         "16xlarge",
+		"r3":         "8xlarge",
+		"r4":         "16xlarge",
+		"t1":         "micro",
+	}
+	platformString := largestInstance[parts[0]]
+
+	if platformString == "" {
+		platformString = "metal" // default
+	}
+
+	// If this is platform instance, return itself
+	if parts[1] == platformString {
+		return instance
+	}
+
+	metalString := fmt.Sprintf("%s.%s", parts[0], platformString)
+
+	// find the metal instance
+	for _, ve := range allInstances {
+		if ve.APIName == metalString {
+			return ve
+		}
+	}
+
+	// in case there is no metal instance we asumme that this instance is platform/metal
+	return instance
+}
+
+func getPlatformVcpu(instance VantageExport, allInstances []VantageExport) string {
+	metalInstance := getMetal(instance, allInstances)
+
+	return strconv.FormatFloat(metalInstance.vCPUs, 'f', -1, 64)
+}
+
+// CPU in the platform
+func getCPUUnits(instance VantageExport) string {
+	// platform_vcpu / nb_vcpu(cpu_name)
+	return "" // TODO @JacobValdemar: Actually attempt to get this value
+}
```

Comment:
@da-ekchajzer I didn't understand the explanation of `CPU.units`. Maybe you can elaborate?

Reply:
In some cases, a "platform" e.i. a metal instance, has two or more CPU (e.g. dual socket motherboard has two CPUs). Unfortunately, the number of CPUs is often not given as such. However, it can be determined simply by dividing the total number of vCPUs in a platform by the number of vCPUs in a CPU. 

Let's take a c5n.metal for exemple. A c5n.metal has 72 vCPUs. The CPU of c5n.metal is [Xeon Platinum 8124M](https://en.wikichip.org/wiki/intel/xeon_platinum/8124m?utm_content=cmp-true) which contains 36 vCPU (same as thread). 

```
nb_cpu = 72 / nb_vcpu(Xeon Platinum 8124M) = 72 / 36 = 2
```
A c5n.metal as 2 CPUs

Is it more clear ?

Reply:
Never mind 😑 I just manually looked up the thread number for each cpu, it that many anyways